src/include/btree_inline.h

```diff
@@ -64,6 +64,21 @@ __wt_evict_page_soon_check(WT_SESSION_IMPL *session, WT_REF *ref, bool *inmem_sp
     btree = S2BT(session);
     page = ref->page;
 
+    if (!__wt_evict_page_is_soon_or_wont_need(page))
+        return (false);
+
+    /*
+     * Don't schedule pages that will require an update-restore eviction. Wait for them to be old
+     * enough to get evicted.
+     */
+    if (F_ISSET_ATOMIC_64(S2C(session), WT_CONN_PRECISE_CHECKPOINT) &&
+      !__wt_evict_aggressive(session) && __wt_page_is_modified(page) &&
+      ((S2C(session)->txn_global.checkpoint_running &&
+         page->modify->newest_commit_timestamp > S2C(session)->txn_global.checkpoint_timestamp) ||
+        (!S2C(session)->txn_global.checkpoint_running &&
+          page->modify->newest_commit_timestamp > S2C(session)->txn_global.stable_timestamp)))
```

Comment:
@quchenhao is there a better/existing version of this check? Which is trying to choose either the checkpoint or stable timestamp depending on whether a checkpoint is active.